### PR TITLE
Basic checks to invest in the project pools

### PR DESCRIPTION
**Pull request by zamith**
Why:

* A user needs to pass some pre requirements in order to invest in each
  of the project pools

This change addresses the need by:

* Adding the checks to the Batch, which means it needs to have access to
  the staking contract.

### CODE REVIEW

File: packages/contracts/contracts/discovery/Staking.sol [hunk 1]
```diff
@@ -63,6 +63,10 @@ contract Staking is IStaking {
         token = _token;
     }
 
+    function hasStaked(address _account) public view returns (bool) {
+        return stakes[_account].total > 0;
```
Comment (naps62) [inline packages/contracts/contracts/discovery/Staking.sol:67]: I think we should only consider `bonded` amounts, no?

**Comment by naps62** [comment]
missing unit tests for these :(

File: packages/contracts/contracts/discovery/Controller.sol [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 import {IProject} from "./interfaces/IProject.sol";
 import {IBatch} from "./interfaces/IBatch.sol";
+import {IStaking} from "./interfaces/IStaking.sol";
 import {Project} from "./Project.sol";
 import {Batch} from "./Batch.sol";
+import {FractalRegistry} from "../fractal_registry/FractalRegistry.sol";
 
 import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
 
+import "hardhat/console.sol";
```
Comment (DavideSilva) [inline packages/contracts/contracts/discovery/Controller.sol:17]: we don't need to import this, right?

**Review by DavideSilva — APPROVED** [review]

File: packages/contracts/contracts/discovery/Controller.sol (hunk 2 above)
Comment (zamith) [reply packages/contracts/contracts/discovery/Controller.sol:17]: Well, don't we? :p

**Pull request merged**